feat(web-analytics): show which properties are optimized for new query engine

lricoy opened the pull request:
## Problem

So far, we have been showing only the properties that are supported by the new query engine, but this is not our final goal, we should educate the users by showing them which filters are optimized so we can get rid of the toggle.

This PR does not remove the query engine toggle, but it stops hiding the unsupported properties when the new query engine is enabled, preferring the optimized sort.

## Changes

- Added a lightning bolt icon next to the supported filters on Web Analytics
- Sort by them first if the user has access to the new engine, so they appear at the front of the list

<img width="663" height="349" alt="image" src="https://github.com/user-attachments/assets/ea21bb8c-0877-4807-9a0b-daa8a5aa1592" />


## How did you test this code?

Manually and automated testing to keep the existing ones working

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

lricoy committed:
feat: show optmized hints

lricoy committed:
chore: extend to include session properties

github-actions[bot] committed:
Update query snapshots

lricoy committed:
chore: add flag to sort only on web analytics

lricoy committed:
chore: conditional sorting

lricoy committed:
fix: support frontend-only allowList

lricoy committed:
fix tests

lricoy committed:
chore: put feature behind flag

lricoy committed:
fix tests

lricoy committed:
fix: woraround kea type limit

lricoy committed:
fix session props sorting

lricoy committed:
tidy up

lricoy committed:
prefer standard asserts

lricoy committed:
rollback unrelated text change

lricoy committed:
review suggestions

lricoy committed:
rename it to supported_by_preaggregated_tables

lricoy committed:
rename all references

lricoy committed:
rename infiniteList prop/logic

lricoy committed:
fix

lricoy committed:
rollback session test changes

lricoy committed:
refactor some tests

jabahamondes commented on posthog/api/test/__snapshots__/test_api_docs.ambr:
```diff
@@ -103,6 +103,7 @@
     '/home/runner/work/posthog/posthog/posthog/session_recordings/session_recording_api.py: Warning [SessionRecordingViewSet]: could not derive type of path parameter "project_id" because model "posthog.session_recordings.models.session_recording.SessionRecording" contained no such field. Consider annotating parameter with @extend_schema. Defaulting to "string".',
     '/home/runner/work/posthog/posthog/posthog/session_recordings/session_recording_playlist_api.py: Warning [SessionRecordingPlaylistViewSet]: could not derive type of path parameter "project_id" because model "posthog.session_recordings.models.session_recording_playlist.SessionRecordingPlaylist" contained no such field. Consider annotating parameter with @extend_schema. Defaulting to "string".' ...
     '/home/runner/work/posthog/posthog/posthog/session_recordings/session_recording_playlist_api.py: Warning [SessionRecordingPlaylistViewSet]: could not derive type of path parameter "session_recording_id" because model "posthog.session_recordings.models.session_recording_playlist.SessionRecordingPlaylist" contained no such field. Consider annotating parameter with @extend_schema. Defaulting to  ...
+    '/home/runner/work/posthog/posthog/posthog/taxonomy/property_definition_api.py: Warning [PropertyDefinitionViewSet > PropertyDefinitionSerializer]: unable to resolve type hint for function "get_supported_by_preaggregated_tables". Consider using a type hint or @extend_schema_field. Defaulting to string.',
```
Should you fix this?

jabahamondes reviewed (approved):
LGTM! not an expert on this topic, but everything looks ok, left some small comments only 🚀

robbie-c commented on frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.tsx:
```diff
@@ -207,20 +208,26 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>([
+                    s.eventOrdering,
+                    s.maxContextOptions,
+                    s.enablePreaggregatedTableHints,
+                ] as any, // workaround as Kea's SelectorTuple has a limit of 11 items: https://github.com/keajs/kea/blob/v3.1.5/src/types.ts#L162-L174
```
I've worked around these before by grouping them up and making another selector just to get a group of props

e.g. you could add a selector that returns {currentTeam, currentProjectId} and then taxonomicGroups can use that selector.

lricoy replied:
This is a nice option. Rafa and Marius updated Kea, and now we can go up to 16 🎉

lricoy commented on frontend/src/scenes/web-analytics/WebPropertyFilters.tsx:
```diff
@@ -10,6 +10,32 @@ import { useState } from 'react'
 
 import { webAnalyticsLogic } from './webAnalyticsLogic'
 
+export const PREAGGREGATED_TABLE_SUPPORTED_PROPERTIES_BY_GROUP = {
```
It won't be required soon so I think we can just drop it now

lricoy committed:
chore: remove kea workaround

lricoy committed:
chore: remove unused parameters and fix imports

lricoy committed:
Merge branch 'master' into feat/show-optmized-hints

lricoy committed:
lint

lricoy committed:
chore: resolve merge conflicts from master

Merged master into feat/show-optmized-hints and resolved conflicts in:
- PropertyFilters.tsx: Combined enablePreaggregatedTableHints and operatorAllowlist props
- InfiniteList.tsx: Combined IconBolt and IconCheck icons, added isActive parameter
- taxonomicFilterLogic.tsx: Combined enablePreaggregatedTableHints and hideBehavioralCohorts props
- WebPropertyFilters.tsx: Added webAnalyticsPropertyAllowList configuration
- mypy-baseline.txt: Merged baseline entries

lricoy committed:
refactor: update some nits

lricoy committed:
Merge branch 'master' into feat/show-optmized-hints

lricoy committed:
Merge branch 'master' into feat/show-optmized-hints

lricoy committed:
chore: update mypy baseline

github-actions[bot] committed:
Update UI snapshots for `chromium` (9)
